x-pack/plugins/security_solution/public/detections/components/rules/data_view_selector/index.tsx

```diff
@@ -11,26 +11,17 @@ import type { EuiComboBoxOptionOption } from '@elastic/eui';
 import { EuiCallOut, EuiComboBox, EuiFormRow, EuiSpacer } from '@elastic/eui';
 
 import type { DataViewListItem } from '@kbn/data-views-plugin/common';
-import type { DataViewBase } from '@kbn/es-query';
 import type { FieldHook } from '../../../../shared_imports';
 import { getFieldValidityAndErrorMessage } from '../../../../shared_imports';
 import * as i18n from './translations';
-import { useKibana } from '../../../../common/lib/kibana';
 import type { DefineStepRule } from '../../../pages/detection_engine/rules/types';
 
 interface DataViewSelectorProps {
-  kibanaDataViews: { [x: string]: DataViewListItem };
+  kibanaDataViews: { [dataViewId: string]: DataViewListItem };
   field: FieldHook<DefineStepRule['dataViewId']>;
-  setIndexPattern: (indexPattern: DataViewBase) => void;
 }
 
-export const DataViewSelector = ({
-  kibanaDataViews,
-  field,
-  setIndexPattern,
-}: DataViewSelectorProps) => {
-  const { data } = useKibana().services;
-
+export const DataViewSelector = ({ kibanaDataViews, field }: DataViewSelectorProps) => {
```

Comment:
I've noticed a missing label for data views controller
<img width="968" alt="Screenshot 2022-07-21 at 11 27 21" src="https://user-images.githubusercontent.com/92328789/180192719-efa54bce-417f-4ea9-a631-820f16024b42.png">

Is it indented?

Reply:
It addressed here: https://github.com/elastic/kibana/pull/136811

Reply:
Yeah I left it out because it felt kind of redundant with the button saying data view and then the label for the dropdown to display data view as well. @nkhristinin has a pr updating the labels anyways [here](https://github.com/elastic/kibana/pull/136811).